### PR TITLE
Client-Server mode benchmark support for infinispan

### PR DESCRIPTION
**Pull request by mylesjao**
The Following is modification and new support in this PR:

* add new infinispan db client with hotrod client to support benchmark for default or specified cache in remote standalone infinispan cluster server. 

* add README for infinispan-binding.

### CODE REVIEW

File: infinispan/src/main/java/com/yahoo/ycsb/db/InfinispanRemoteClient.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   
+   @Override
+   public void cleanup() {
+      remoteIspnManager.stop();
+      remoteIspnManager = null;
+   }
+   
+   @Override
+   public int insert(String table, String recordKey, HashMap<String, ByteIterator> values) {
+	   String compositKey = createKey(table, recordKey);
```
Comment (busbey) [inline infinispan/src/main/java/com/yahoo/ycsb/db/InfinispanRemoteClient.java]: indentation in this file is inconsistent, could you fix it please?

File: infinispan/src/main/java/com/yahoo/ycsb/db/InfinispanRemoteClient.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.Vector;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
```
Comment (busbey) [inline infinispan/src/main/java/com/yahoo/ycsb/db/InfinispanRemoteClient.java]: What is the infinispan logging util package? Does it offer some compelling difference over a more common option like slf4j or commons-logging?

File: infinispan/src/main/java/com/yahoo/ycsb/db/RemoteCacheManagerHolder.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+
+public class RemoteCacheManagerHolder {
+	
+	private static RemoteCacheManager cacheManager = null;
+	
+	private RemoteCacheManagerHolder() {}
+	
+	public static RemoteCacheManager getInstance(Properties props){
```
Comment (busbey) [inline infinispan/src/main/java/com/yahoo/ycsb/db/RemoteCacheManagerHolder.java]: this doesn't ensure only a single RemoteCacheManager is made and it unsafely publishes cacheManager before it is initialized.

either synchronize the getInstance method or use a proper double-check idiom and make cacheManager volatile.

**Comment by mylesjao** [comment]
Thanks for your refactoring. I was too busy last month and missed all notifications...

**Comment by busbey** [comment]
no worries! if you could take a few minutes to test out the changes on #309 it would help get it merged in.

**Comment by mylesjao** [comment]
I've pulled [#309](https://github.com/brianfrankcooper/YCSB/pull/309)  and run benchmark on infinispan-server-7.2.2.Final with all workloads. everything works well and without any error.

**Comment by mylesjao** [comment]
:+1:

**Pull request closed without merging**